### PR TITLE
fix event sorting issue

### PR DESCRIPTION
**Pull request by damonsharp**
Fixes sorting issue with events causing nav buttons to be disabled + grunt build.

### CODE REVIEW

File: includes/abstracts/calendar.php [hunk 1]
```diff
@@ -284,8 +284,8 @@ public function __construct( $calendar, $view = '' ) {
 
 			// Set earliest and latest event timestamps.
 			if ( $this->events && is_array( $this->events ) ) {
-				$this->earliest_event = intval( current( array_keys( $this->events ) ) );
-				$this->latest_event   = intval( key( array_slice( $this->events, -1, 1, true ) ) );
+				$this->earliest_event = intval( current( $this->events )[0]->end );
```
Comment (zachwills) [inline includes/abstracts/calendar.php]: Is this `[0]` gauaranteed to always be set?

Comment (zachwills) [reply includes/abstracts/calendar.php]: Does this snippet `current()[0]` work in the minimum required PHP version for the plugin? I know in older PHP versions you couldn't access an array directly after a function.

Comment (damonsharp) [reply includes/abstracts/calendar.php]: @zachwills `[0]` should always be there as it's coming directly from the Google calendar feed data, but a check is always good, and I should have put that in. We were requiring PHP 5.5, but that was with the upgraded googleapi lib, and was rolled back, so I'll double check in 5.4, and mod as necessary.

**Comment by damonsharp** [comment]
@zachwills code refactored to support PHP 5.3+ (the plugin's version requirement). should be ready for another review.

**Comment by damonsharp** [comment]
@zachwills was using the end date as the start date. give this a go.

**Pull request merged**